src/core/src/op/scatter_update.cpp

```diff
@@ -2,134 +2,115 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "ngraph/op/scatter_update.hpp"
+#include "openvino/op/scatter_update.hpp"
 
 #include "bound_evaluate.hpp"
 #include "itt.hpp"
-#include "ngraph/shape.hpp"
-#include "ngraph/type/element_type.hpp"
-#include "ngraph/type/element_type_traits.hpp"
-#include "ngraph/validation_util.hpp"
+#include "openvino/core/shape.hpp"
+#include "openvino/core/type/element_type.hpp"
+#include "openvino/core/type/element_type_traits.hpp"
+#include "openvino/op/constant.hpp"
 #include "openvino/reference/scatter_update.hpp"
-
-using namespace std;
-using namespace ngraph;
-
-op::v3::ScatterUpdate::ScatterUpdate(const Output<Node>& data,
-                                     const Output<Node>& indices,
-                                     const Output<Node>& updates,
-                                     const Output<Node>& axis)
+#include "validation_util.hpp"
+
+namespace ov {
+namespace op {
+namespace v3 {
+ScatterUpdate::ScatterUpdate(const Output<Node>& data,
+                             const Output<Node>& indices,
+                             const Output<Node>& updates,
+                             const Output<Node>& axis)
     : util::ScatterBase(data, indices, updates, axis) {}
 
-shared_ptr<Node> op::v3::ScatterUpdate::clone_with_new_inputs(const OutputVector& new_args) const {
+std::shared_ptr<Node> ScatterUpdate::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v3_ScatterUpdate_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<v3::ScatterUpdate>(new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3));
+    return std::make_shared<ScatterUpdate>(new_args.at(0), new_args.at(1), new_args.at(2), new_args.at(3));
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace scatter_update {
-namespace {
-template <element::Type_t ET>
-std::vector<int64_t> get_indices(const HostTensorPtr& in) {
-    auto data_ptr = in->get_data_ptr<ET>();
-    return std::vector<int64_t>(data_ptr, data_ptr + in->get_element_count());
-}
-}  // namespace
-}  // namespace scatter_update
-
-#define GET_INDICES(a, ...)                                                                   \
-    case element::Type_t::a: {                                                                \
-        OV_OP_SCOPE(OV_PP_CAT3(get_scatter_update_indices, _, a));                            \
-        indices_casted_vector = scatter_update::get_indices<element::Type_t::a>(__VA_ARGS__); \
-    } break;
+bool ScatterUpdate::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
+    OV_OP_SCOPE(v3_ScatterUpdate_evaluate);
+    OPENVINO_ASSERT(inputs.size() == 4);
```

Comment:
Just to consider to use `shape_infer`, but is much complex as do more detailed validation.

Reply:
As agreed offline, "stronger" version of axis normalization is enough to cover safety issues.